zebedee-cms/src/main/java/com/github/onsdigital/zebedee/ZebedeeConfiguration.java

```diff
@@ -151,301 +148,434 @@ public ZebedeeConfiguration(Path rootPath, boolean enableVerificationAgent) thro
+        return this.usersService;
+    }
+
+    public VerificationAgent getVerificationAgent() {
+        return verificationAgent;
+    }
+
+    public DatasetService getDatasetService() {
+        return datasetService;
+    }
+
+    public ImageService getImageService() {
+        return imageService;
+    }
+
+    public KafkaService getKafkaService() {
+        return kafkaService;
+    }
 
-        // TODO: Remove after migration to JWT sessions is complete
+    public ServiceStore getServiceStore() {
+        return serviceStore;
+    }
+
+    public CollectionKeyring getCollectionKeyring() {
+        return collectionKeyring;
+    }
+
+    public EncryptionKeyFactory getEncryptionKeyFactory() {
+        return encryptionKeyFactory;
+    }
+
+    public CollectionKeyCache getSchedulerKeyringCache() {
+        return schedulerKeyCache;
+    }
+
+    public StartUpNotifier getStartUpNotifier() {
+        return startUpNotifier;
+    }
+
+    public Notifier getSlackNotifier() {
+        return slackNotifier;
+    }
+
+    /**
+     * Initialise the teams service.
+     *
+     * @param zebedeePath                       the zebedee base directory (i.e. `$zebedee_root/zebedee`)
+     * @param permissionsServiceServiceSupplier a {@link ServiceSupplier} to supply the {@link PermissionsService} without
+     *                                          creating a circular dependency
+     * @throws IOException If a filesystem error occurs.
+     *
+     * TODO: Remove after migration to JWT sessions is complete
+     */
+    private void initTeamsService(Path zebedeePath,
+                                  ServiceSupplier<PermissionsService> permissionsServiceServiceSupplier)
+            throws IOException {
+
+        Path teamsPath = createDir(zebedeePath, TEAMS);
         if (cmsFeatureFlags().isJwtSessionsEnabled()) {
             this.teamsService = new StubbedTeamsServiceImpl();
+            info().log("JWT sessions enabled: stubbed teams service initialised");
         } else {
             this.teamsService = new TeamsServiceImpl(
-                    new TeamsStoreFileSystemImpl(teamsPath), this::getPermissionsService);
-        }
+                    new TeamsStoreFileSystemImpl(teamsPath), permissionsServiceServiceSupplier);
 
-        this.published = createPublished();
+            info().data("teams_path", teamsPath.toString())
+                    .log("legacy teams service initialised");
+        }
+    }
 
+    /**
+     * Initialise the permissions service.
+     *
+     * @param zebedeePath the zebedee base directory (i.e. `$zebedee_root/zebedee`)
+     * @param teamsService the {@link TeamsService}
+     * @throws IOException If a filesystem error occurs.
+     */
+    private void initPermissionsService(Path zebedeePath, TeamsService teamsService) throws IOException {
+        Path permissionsPath = createDir(zebedeePath, PERMISSIONS);
         initialisePermissions(permissionsPath);
```

Comment:
I know this is how you found it but can we rename this to something like `initialiseAccessMapping(permissionsPath)` to make it clearer what this does and to differentiate it from the method we're already in.